### PR TITLE
chore: fix some lint warnings

### PR DESCRIPTION
**Pull request by ST-DDT**

### CODE REVIEW

**Review by prisis — DISMISSED** [review]

File: src/helpers.ts [hunk 1]
```diff
@@ -342,7 +342,7 @@ export class Helpers {
     const RANGE_REP_REG = /(.)\{(\d+)\,(\d+)\}/;
     const REP_REG = /(.)\{(\d+)\}/;
     const RANGE_REG = /\[(\d+)\-(\d+)\]/;
-    let min, max, tmp, repetitions;
+    let min, max, tmp, repetitions: number;
```
Comment (Shinigami92) [inline src/helpers.ts:345]: We should split this to separate lines

File: test/fake.spec.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ describe('fake', () => {
 
     it('does not allow undefined parameters', () => {
       expect(() =>
-        // @ts-expect-error
+        // @ts-expect-error The parameter is required
```
Comment (Shinigami92) [inline test/fake.spec.ts:33]:
```suggestion
        // @ts-expect-error: The parameter is required
```

File: src/helpers.ts (hunk 1 above)
Comment (ST-DDT) [reply src/helpers.ts:345]: Do you have time to fix these?

File: src/helpers.ts [hunk 3]
```diff
@@ -434,9 +437,9 @@ export class Helpers {
    * faker.helpers.uniqueArray(faker.definitions.name.first_name, 6)
    * faker.helpers.uniqueArray(["Hello", "World", "Goodbye"], 2)
    */
-  uniqueArray<T>(source: T[] | (() => T), length: number): T[] {
+  uniqueArray<T>(source: readonly T[] | (() => T), length: number): T[] {
     if (Array.isArray(source)) {
-      const set = new Set(source);
+      const set = new Set<T>(source);
       const array = Array.from(set);
       return this.faker.helpers.shuffle(array).splice(0, length);
```
Comment (Shinigami92) [inline src/helpers.ts:444]: Just a note for a later PR, IMO we should use `slice` instead of `splice` here :thinking:

**Review by prisis — APPROVED** [review]

**Pull request merged**